### ISSUE TITLE
Error signing Taproot inputs on Ledger devices

### ISSUE THREAD

**Issue by craigraw** [bitcoin-core/HWI#608]
When signing transactions from a Taproot wallet on Ledger (both mainnet and testnet), the device accepts the PSBT provided and permits the user a full confirmation of the transaction details (including fee). On accepting the transaction however, the device returns the following error via HWI:
```
('0x6a82', 'Error in <BitcoinInsType.SIGN_PSBT: 4> command', '')
```

Example PSBT: [Ledger Taproot.psbt.txt](https://github.com/bitcoin-core/HWI/files/8772057/Ledger.Taproot.psbt.txt) described as follows:
```
Transaction with txid: 58547f7b775fa28d0f0326186dde17e30dc422e5da2b1bf3a7b9c732543d3e6d version 2 size 125 locktime 0
 Transaction input references txid: c3de8c30767ae3d5ef447f5bca26404e9e35845cc8c8081e458524e588a0b522 vout 0 with script 
 Transaction output value: 633 to addresses [tb1pu5v9r7hkczeymg4lzt5xwj3ky7aa5jnq0kem528dktz6wyhv5t0s76k9fs] with script hex 5120e51851faf6c0b24da2bf12e8674a3627bbda4a607db3ba28edb2c5a712eca2df to script OP_1 e51851faf6c0b24da2bf12e8674a3627bbda4a607db3ba28edb2c5a712eca2df
 Transaction output value: 454 to addresses [tb1qjmkgkp3geqd5ud2wtw7caga5ufdyzhw3u2wcxp] with script hex 001496ec8b0628c81b4e354e5bbd8ea3b4e25a415dd1 to script OP_0 96ec8b0628c81b4e354e5bbd8ea3b4e25a415dd1
Pubkey with master fingerprint 55d489f5 at path m/86'/1'/0': tpubDCRgVmGZrWPc14FH4wTv32f21mZiuQVfAyN2cNMthfY5Jrshty8dv8HZQTsF5oD58TWESoiMXNoLzWLzn8Sh3PSxXECdytzSExbt65mgGnX
Found input non witness utxo with txid: c3de8c30767ae3d5ef447f5bca26404e9e35845cc8c8081e458524e588a0b522 version 2 size 125 locktime 2227563
 Transaction input references txid: 8aa27b96d32862ee06761f1e0f5e9ff1a6f771f7e300602727a6f44132d1f6db vout 1 with script 
 Transaction output value: 1230 to address tb1pq57fzl9cask9pkx840v3z9h24wjegfzkasztv60904htudj3tedqq7vtrf with script hex 5120053c917cb8ec2c50d8c7abd91116eaaba5942456ec04b669e57d6ebe36515e5a to script OP_1 053c917cb8ec2c50d8c7abd91116eaaba5942456ec04b669e57d6ebe36515e5a
 Transaction output value: 141617 to address tb1qckta2zvxceplm7xdnhu86tkaynxm0zdzmlluzs with script hex 0014c597d50986c643fdf8cd9df87d2edd24cdb789a2 to script OP_0 c597d50986c643fdf8cd9df87d2edd24cdb789a2
Found input witness utxo amount 1230 script hex 5120053c917cb8ec2c50d8c7abd91116eaaba5942456ec04b669e57d6ebe36515e5a script OP_1 053c917cb8ec2c50d8c7abd91116eaaba5942456ec04b669e57d6ebe36515e5a addresses [tb1pq57fzl9cask9pkx840v3z9h24wjegfzkasztv60904htudj3tedqq7vtrf]
Found input sighash_type All (Taproot)
Found input taproot key derivation for key 22bbb6d637f826764d7caa9312b969b1e0e92e62804dfc6afd9f76aaa2dbd98b with master fingerprint 55d489f5 at path m/86'/1'/0'/0/0
Found input taproot internal key 22bbb6d637f826764d7caa9312b969b1e0e92e62804dfc6afd9f76aaa2dbd98b
Found output taproot key derivation for key e45700456e1d852df15c3cb01c408f2087c2c8ee4bacf4be28af3ba8f7004727 with master fingerprint 55d489f5 at path m/86'/1'/0'/1/0
Found output taproot internal key e45700456e1d852df15c3cb01c408f2087c2c8ee4bacf4be28af3ba8f7004727
```

This is with HWI 2.1.1, appears to be platform independent, and occurs on both the Nano X and Nano S Plus.
Similar PSBTs are being successfully signed on the Trezor One and T. 

PS I did try to test this with Speculos, but was unable to get the Bitcoin 2.0.4 app running. The 2.0.1 emulation returned the PSBT unsigned without any onscreen activity.

**Comment by achow101** [comment]
The test suite passes with 2.0.1 through 2.0.3 but starts failing with 2.0.4. There's nothing in the [changelog](https://github.com/LedgerHQ/app-bitcoin-new/blob/develop/CHANGELOG.md) that stands out as causing the break. Will debug further.

**Comment by achow101** [comment]
@craigraw Does this PSBT happen to not belong to the device(s) you are trying to sign with? I tthink 2.0.4 may have changed something about how external inputs are handled.

**Comment by achow101** [comment]
Hmm, I think the problem I am debugging is different from the one in this issue. The one I am seeing is with status `0xb007` but this is `0x6a82`. @craigraw you are certain that you used HWI 2.1.1?

**Comment by craigraw** [comment]
> @craigraw Does this PSBT happen to not belong to the device(s) you are trying to sign with?

It definitely belongs to the device. The code creating it is the same as that creating the PSBTs for Trezor.

> @craigraw you are certain that you used HWI 2.1.1?

Yes. Sparrow v1.6.5 (released yesterday) contains HWI 2.1.1, and I have recreated the error on both MacOS and Ubuntu.

**Comment by achow101** [comment]
I was able to replicate the issue. This is an issue with the way that ledger handles the sighash type field. It appears that it does not recognize SIGHASH_DEFAULT (which is what your PSBT has).

I have opened an issue with Ledger: https://github.com/LedgerHQ/app-bitcoin-new/issues/47
